### PR TITLE
Add angular-eslint rules

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds Angular-specific lint rules to MSAL to lint msal-angular.

See [angular-eslint](https://github.com/angular-eslint/angular-eslint) for more details. [Base](https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/configs/base.json) and [recommended rules](https://github.com/angular-eslint/angular-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json) added.

### CODE REVIEW

**Comment by sameerag** [comment]
> Coverage increased (+10.8%) to 94.343%

Way to go @jo-arroyo

**Review by tnorling — APPROVED** [review]

**Review by sameerag — APPROVED** [review]
Lgtm

File: .eslintrc.json [hunk 1]
```diff
@@ -68,7 +69,23 @@
         "react/no-string-refs": 2,
         "react/prefer-stateless-function": 2,
         "react/self-closing-comp": 2,
-        "react/sort-comp": 2
+        "react/sort-comp": 2,
+        "@angular-eslint/component-class-suffix": 2,
```
Comment (jasonnutter) [inline .eslintrc.json:73]: Can these be moved to the top of the list (alphabetical order)?

**Review by jasonnutter — APPROVED** [review]
One thing otherwise looks good!

File: .eslintrc.json (hunk 1 above)
Comment (jo-arroyo) [reply .eslintrc.json]: Makes sense. Moved.

**Pull request merged**